hcert_spec.md

```diff
@@ -165,17 +177,76 @@ However, regardless if an Issuer decides to use HSMs or not, a key roll-over sch
 This specification may be used in a way which implies receiving data from untrusted sources into systems which may be of mission-critical nature. To minimise the risks associated with this attack vector, all input fields MUST be properly validated by data types, lengths and contents. The Issuer Signature SHALL also be verified before any processing of the contents of the EHC takes place. However, the validation of the Issuer Signature implies parsing the Protected Issuer Header first, in which a potential attacker may attempt to inject carefully crafted information designed to compromise the security of the system.
 
 
-## Appendix A
+# Appendix A - Payload
 
 A proposed payload schema for [EU Health Certficate v1](eu_hcert_v1_schema.yaml).
 
+# Appendix B - Trust management
+
+The signature on the EHC requires a public key to verify. Countries, or institutions within countries, need to place those signatures. And ultimately every verifier needs to have a list of the public keys it is willing to trust (the public key is not part of the signature).
+
+For this a simplified variation on the ICAO "_Master list_: will be used, tailored to this health application. Where each country is ultimately responsible for compiling their own master list. But with the aid of a coordinating secretariat for operational and practical purposes.
+
+The system consists of (just) two layers; for each Member State one or more country level certificate that each sign one or more document signing certificates that are used in day to day operations.
+
+The member-state certificates are called Certificate Signer Certificate Authority (CSCA) certificates and are (typically) self signed certificates. Countries may have more than one (e.g. in case of regional devolution). 
+
+Memberstates are required to keep a public register of these certificates at a stable URL.
+
+Memberstates may then bilaterally exchange CSCA certificates with a number of other States, verify these bilaterally and thus compile their own lists of CSCA certificates: a (MS specific) Master List. 
+
+These CSCA certificates regularly sign the Document Signing Certificates (DSC) used in day to day operations. Memberstates will each will maintain a public register of the DSC certificates that is kept current.
+
+Other memberstates must regularly fetch these list of DSC certificates and cryptographically verify these against the CSCA certificates (that they have verified by other, non-digital, means).  
+
+The resulting list of DSC certificates then provides the acceptable public keys (and the corresponding KIDs) that verifiers can use to validate the signature on the CWT in the Qr code. 
+
+Verifiers should fetch update so this list regularly.
```

Comment:
It would be good to explicitly state that the file format of the trusted list may vary, e.g. it can be a plain JWKS like https://github.com/ehn-digital-green-development/hcert-testdata/blob/main/testdata/jwks.json

Reply:
Adding.